File: app/src/main/java/com/mapmyindia/sdk/demo/java/utils/CheckInternet.java
package com.mapmyindia.sdk.demo.java.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * Created by CEINFO on 26-02-2019.
 */
public class CheckInternet {
    /* function to check internet connection*/
    public static boolean isNetworkAvailable(Context mContext) {
        ConnectivityManager connectivityManager
                = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null;
    }
}

File: app/src/main/java/com/mmi/sdk/demo/kotlin/utility/CheckInternet.kt
package com.mmi.sdk.demo.kotlin.utility

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

/**
 * Created by CEINFO on 26-02-2019.
 */
class CheckInternet {
    companion object {
        private fun isNetworkAvailable(ctx: Context): Boolean {
            val connectivityManager = ctx.getSystemService(Context.CONNECTIVITY_SERVICE)
            return if (connectivityManager is ConnectivityManager) {
                val networkInfo: NetworkInfo? = connectivityManager.activeNetworkInfo
                networkInfo?.isConnected ?: false
            } else false
        }
    }

}